src/middlewared/middlewared/utils/cpu.py

```diff
@@ -137,3 +141,53 @@ def amd_cpu_temperatures(amd_metrics: dict) -> dict:
             return dict(enumerate([amd_sensors['temp1']] * core_count))
         elif 'temp1_input' in amd_sensors['temp1']:
             return dict(enumerate([amd_sensors['temp1']['temp1_input']] * core_count))
+
+
+def read_cpu_temps() -> dict:
+    """
+    Read CPU temperatures using libsensors.
+    Returns data in the format expected by existing temperature processing functions.
+
+    Returns:
+        Dictionary with chip names as keys and temperature readings as nested dicts
+        Example: {'coretemp-isa-0000': {'Core 0': 48.0}, 'k10temp-pci-00c3': {'Tctl': 67.0}}
+    """
+    with contextlib.suppress(OSError, RuntimeError):
+        with SensorsWrapper() as sensors:
```

Comment:
By using the context manager, it's doing a bunch of CDDL setup functions each time it's called which is painful and unnecessary. We should instantiate a `sensors = SensorsWrapper()` object once and then just do `sensors.get_cpu_temperatures()`